pkg/driver/trace_driver.go

```diff
@@ -375,11 +603,11 @@ func (d *Driver) functionsDriver(list *list.List, announceFunctionDone *sync.Wai
 	waitForInvocations.Wait()
 
 	log.Debugf("All the invocations for function %s have been completed.\n", function.Name)
-	announceFunctionDone.Done()
-
 	atomic.AddInt64(totalSuccessful, successfulInvocations)
 	atomic.AddInt64(totalFailed, failedInvocations)
 	atomic.AddInt64(totalIssued, numberOfIssuedInvocations)
+	atomic.AddInt64(entriesWritten, functionsInvoked)
+	announceFunctionDone.Done()
```

Comment:
Why has the announceFunctionDone.Done() code been moved from its previous position on line 378?